server/server.go

```diff
@@ -362,6 +364,26 @@ func NewServer(opts *Options) (*Server, error) {
 		httpReqStats: make(map[string]uint64), // Used to track HTTP requests
 	}
 
+	if opts.TLSRateLimit > 0 {
+		s.connRateCounter = newRateCounter(opts.tlsConfigOpts.RateLimit)
+		s.startGoRoutine(func() {
```

Comment:
This would not work since the server has not been started yet. `startGoRoutine()` will not start the go routine if `s.grRunning` is not set to true, which happens only in `Server.Start()`.
Since this routine just does something based on a ticker, we could arguably change that to a timer func, but it would need to be stopped on Shutdown().
If you keep this go routine, still move the s.startGoRoutine() to Start(), or even AcceptLoop() since this applies only to client (when TLS is enabled).
(also, make it a separate function, so that in AcceptLoop we have `s.startGoRoutine(s.logRejectedTLSConns)` or something like that..)